File: test_avatar.py
import requests
import string
import random
import os
import time

base_url = "http://188.166.65.112:8080"
email = ''.join(random.choices(string.ascii_lowercase, k=10)) + "@test.com"
password = "password123"

session = requests.Session()

print(f"Registering {email}...")
res = session.post(f"{base_url}/api/auth/register", json={"email": email, "password": password, "fullName": "Test User"})
print(res.status_code, res.text)

print("Confirming email in db...")
os.system(f'psql "postgresql://nexclone:devpassword123!@188.166.65.112:5432/nexclone_dev" -c "UPDATE \\"AspNetUsers\\" SET \\"EmailConfirmed\\" = true, \\"IsVerified\\" = true WHERE \\"Email\\" = \'{email}\';"')

print("Logging in...")
res = session.post(f"{base_url}/api/auth/login", json={"email": email, "password": password})
print(res.status_code, res.text)

token = res.json().get("token")
session.headers.update({"Authorization": f"Bearer {token}"})

# Add credits to Wallets via User ID
os.system(f'psql "postgresql://nexclone:devpassword123!@188.166.65.112:5432/nexclone_dev" -c "INSERT INTO \\"UserWallets\\" (\\"UserId\\", \\"WalletTypeId\\", \\"Balance\\") VALUES ((SELECT \\"Id\\" FROM \\"AspNetUsers\\" WHERE \\"Email\\" = \'{email}\'), 1, 100);" || true')

print("Testing start-avatar...")
with open('scratch/face.jpg', 'rb') as f_img, open('scratch/tiny.wav', 'rb') as f_audio:
    files = {
        'Image': ('face.jpg', f_img.read(), 'image/jpeg'),
        'Audio': ('tiny.wav', f_audio.read(), 'audio/wav')
    }
    data = {'prompt': 'Testing E2E avatar'}

    res = session.post(f"{base_url}/api/video/start-avatar", files=files, data=data)
print("Status Code:", res.status_code)
print("Response:", res.text)

if res.status_code == 200:
    task_id = res.json().get('taskId')
    print("Got taskId:", task_id)
    print("Checking status...")
    for _ in range(300):
        time.sleep(10)
        res = session.get(f"{base_url}/api/video/status/{task_id}")
        print("Status:", res.text)
        if "succeeded" in res.text or "failed" in res.text:
            break

File: scratch/test-avatar.js
const crypto = require('crypto');
const { execSync } = require('child_process');
const fs = require('fs');

const BASE_URL = 'http://localhost:8080';

async function fetchApi(path, options = {}) {
  const url = `${BASE_URL}${path}`;
  const res = await fetch(url, options);
  
  let text = await res.text();
  let data;
  try { data = JSON.parse(text); } catch { data = text; }
  
  return {
    status: res.status,
    headers: res.headers,
    data
  };
}

async function runTests() {
  console.log("=== Avatar Video Test ===");
  
  const testEmail = `testuser_${crypto.randomBytes(4).toString('hex')}@example.com`;
  const password = "Password123!";
  
  console.log(`\n1. Registering user: ${testEmail}`);
  const regRes = await fetchApi("/api/auth/register", {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      fullName: "Test User",
      email: testEmail,
      password: password,
      country: "US"
    })
  });
  
  console.log(`\n-> Setting up user in DB...`);
  const psqlPrefix = `docker exec -e PGPASSWORD=localpassword nexclone-postgres psql -U postgres -d nexclonedb -t -c`;
  
  // Verify
  execSync(`${psqlPrefix} "UPDATE \\\"AspNetUsers\\\" SET \\\"IsVerified\\\" = true WHERE \\\"Email\\\" = '${testEmail}';"`);
  
  // Get UserId
  const userIdRaw = execSync(`${psqlPrefix} "SELECT \\\"Id\\\" FROM \\\"AspNetUsers\\\" WHERE \\\"Email\\\" = '${testEmail}';"`);
  const userId = userIdRaw.toString().trim();
  
  // Insert a Plan (or find an existing one)
  const planIdRaw = execSync(`${psqlPrefix} "SELECT \\\"Id\\\" FROM \\\"Plans\\\" LIMIT 1;"`);
  let planId = planIdRaw.toString().trim();
  
  // Insert a Subscription
  execSync(`${psqlPrefix} "INSERT INTO \\\"Subscriptions\\\" (\\\"UserId\\\", \\\"PlanId\\\", \\\"Status\\\", \\\"StartDate\\\", \\\"EndDate\\\", \\\"CreatedAt\\\") VALUES ('${userId}', ${planId}, 'active', NOW(), NOW() + INTERVAL '30 days', NOW());"`);
  
  // Update Wallet
  execSync(`${psqlPrefix} "UPDATE \\\"UserWallets\\\" SET \\\"Balance\\\" = 1000 WHERE \\\"UserId\\\" = '${userId}';"`);
  
  console.log(`\n3. Logging in...`);
  const loginRes = await fetchApi("/api/auth/login", {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({
      email: testEmail,
      password: password
    })
  });
  
  const setCookieHeader = loginRes.headers.get('set-cookie');
  const cookies = setCookieHeader ? setCookieHeader.split(', ').find(c => c.includes('jwt=')) : null;
  const authCookie = cookies ? cookies.split(';')[0] : '';
  
  const meRes = await fetchApi("/api/auth/me", {
    headers: { Cookie: authCookie }
  });
  
  const user = meRes.data;
  const activeSubs = user.activeSubscriptions;
  const subId = activeSubs[0].id;
  
  console.log(`\n4. Testing POST /api/video/start-avatar...`);
  const axios = require('axios');
  const FormData = require('form-data');
  const formData = new FormData();
  
  const imgData = fs.readFileSync('/root/nexmedia/nexclone/lenna.jpg');
  formData.append('image', imgData, { filename: 'lenna.jpg', contentType: 'image/jpeg' });
  
  const audioData = fs.readFileSync('/root/nexmedia/nexclone/dummy.mp3');
  formData.append('audio', audioData, { filename: 'dummy.mp3', contentType: 'audio/mpeg' });
  
  formData.append('prompt', "The speaker talks naturally");
  formData.append('renderingSpeed', "std");
  formData.append('subscriptionId', subId);
  
  try {
    const avatarRes = await axios.post("http://localhost:8080/api/video/start-avatar", formData, {
      headers: {
        Cookie: authCookie,
        ...formData.getHeaders()
      }
    });
    console.log(`Avatar Status: ${avatarRes.status}`);
    console.log(`Avatar Response:`, avatarRes.data);
  } catch (err) {
    console.log(`Avatar Error Status: ${err.response?.status}`);
    console.log(`Avatar Error Data:`, err.response?.data);
  }
}
runTests();
